### PR TITLE
Modernize plugin jenkins-ci.plugin.version + more

### PR DESCRIPTION
**Pull request by justinharringa**
See #7, #8, #9, #10 from dependabot. Additional bumps for Enforcer rules

### CODE REVIEW

**Comment by justinharringa** [comment]
Closes #7 #8 #9 #10

**Comment by justinharringa** [comment]
This should also clear up the CI error in #6

**Comment by justinharringa** [comment]
Closes #13

**Comment by justinharringa** [comment]
Closes #17

File: pom.xml [hunk 1]
```diff
@@ -3,7 +3,7 @@
   <parent>
     <groupId>org.jenkins-ci.plugins</groupId>
     <artifactId>plugin</artifactId>
-    <version>2.11</version>
```
Comment (superfell) [inline pom.xml]: Does this affect the minimum required version of Jenkins?

File: pom.xml [hunk 2]
```diff
@@ -48,16 +48,17 @@
   </scm>
   
   <properties>
-    <commons-httpclient.version>3.1</commons-httpclient.version>
-    <commons-io.version>2.4</commons-io.version>
-    <commons-codec.version>1.4</commons-codec.version>
+    <commons-httpclient.version>3.1-jenkins-1</commons-httpclient.version>
```
Comment (superfell) [inline pom.xml]: I wasn't able to find any information about this version, do you know what the deal with it is?

File: pom.xml (hunk 1 above)
Comment (justinharringa) [reply pom.xml]: Nope but this does: https://github.com/jenkinsci/chatter-notifier-plugin/pull/11/files#diff-600376dffeb79835ede4a0b285078036R58

File: pom.xml (hunk 2 above)
Comment (justinharringa) [reply pom.xml]: Came from #7 but essentially looks like Jenkins forked this to backport security fixes https://github.com/jenkinsci/lib-commons-httpclient/commits/master

**Comment by justinharringa** [comment]
@superfell did my answers make sense?

**Comment by justinharringa** [comment]
Bump to `jackson-mapper-asl:1.9.13-atlassian-1` closes #12

**Comment by superfell** [comment]
> @superfell did my answers make sense?

Yes, thanks!

**Pull request merged**